core/ecschema-metadata/src/Metadata/Unit.ts

```diff
@@ -63,6 +63,19 @@ export class Unit extends SchemaItem {
     return true;
   }
 
+  /**
+   * Returns true if a conversion can be calculated between the input units
+   * @alpha
+   */
+  public static areCompatibleSync(unitA: Unit, unitB: Unit): boolean {
+    const unitAPhenomenon = unitA.phenomenon;
+    const unitBPhenomenon = unitB.phenomenon;
+
+    if (!unitAPhenomenon || !unitBPhenomenon || !unitAPhenomenon.matches(unitBPhenomenon))
+      return false;
+    return true;
+  }
+

```

Comment:
because this method only needs the keys we can have a single implementation where the async calls the sync method.  The LazyLoadedPhenomenon has all the information needed to see if their key matches, so the async call is actually wrong to await.